gtsam/geometry/triangulation.h

```diff
@@ -494,5 +496,9 @@ TriangulationResult triangulateSafe(const CameraSet<CAMERA>& cameras,
     }
 }
 
+// Vector of Cameras - used by the Python/MATLAB wrapper
+using CameraSetCal3Bundler = CameraSet<PinholeCamera<Cal3Bundler>>;
+using CameraSetCal3_S2 = CameraSet<PinholeCamera<Cal3_S2>>;
```

Comment:
I understand why you need this.

@ProfFan  seems like the wrapper does not maintain ordering of declarations, so even though the typedef exists in `gtsam.i`, these aliases are required to avoid `undefined` errors.

Reply:
I spoke to @swarrier246 offline and we figured out how to not use the `using` alias just for the sake of the wrapper.

Reply:
I think that is very strange as typedefs should be generated on the *top*. BTW, I thought we are ignoring the CameraSet for better performance, no?

Reply:
We're keeping the `using `. How would ignoring CameraSet improve performance? Won't it break triangulatepoint3? That function expects a CameraSet

Reply:
I think it will not as ignore will only skip wrapping the CameraSet class, the function signature is still there. I suggest you look at the generated C++ code and do a diff to understand what is going on there.

Reply:
I looked into the generated cpp and the typedefs are definitely not generated at the top or in sequence. There are cases where a type is used even before the typedef declaration has been made known, resulting in the wrapper throwing `undefined reference` errors.

I think @ProfFan and @swarrier246 are not on the same page. Fan is referring to updating the preamble.h so that CameraSet is treated as a Python list in the wrapper. Honestly, I don't like that design since we need to touch multiple files to incorporate it, plus a python list accepts all kinds of objects while the CameraSet should only accept specific camera models. Converting to a list kills the type information and can result in potential bugs and errors.

Reply:
I kinda agree with @varunagrawal 's reasoning on this one. @ProfFan is it a significant performance issue?

Reply:
I think Varun you have a misunderstanding here. *Binding as list* does not make it a Python `list`. You still need to manually create a `CameraSet` object, just that object has a `list` like interface and can be used as if it is a list. You can try putting in something that is not a `Camera`, and it will fail.

Reply:
Ah I see. Then yeah if we get the performance boost, it makes sense to do that. Then we don't need to add all the STL vector methods in the wrapper like it is being currently done, just the class definition should be enough.

@swarrier246 really sorry about this, but I agree with Fan on reverting back to the preamble.h and CMake updates.